lib/ts/recipe/openid/recipe.ts

```diff
@@ -0,0 +1,128 @@
+            override: this.config.override.jwtFeature,
+        });
+
+        let builder = new OverrideableBuilder(RecipeImplementation(this.config, this.jwtRecipe.recipeInterfaceImpl));
+
+        this.recipeImplementation = builder.override(this.config.override.functions).build();
+
+        let apiBuilder = new OverrideableBuilder(APIImplementation());
+
+        this.apiImpl = apiBuilder.override(this.config.override.apis).build();
+    }
+
+    static getInstanceOrThrowError(): OpenIdRecipe {
+        if (OpenIdRecipe.instance !== undefined) {
+            return OpenIdRecipe.instance;
+        }
+        throw new Error("Initialisation not done. Did you forget to call the SuperTokens.init function?");
+    }
+
+    static init(config?: TypeInput): RecipeListFunction {
+        return (appInfo, isInServerlessEnv) => {
+            if (OpenIdRecipe.instance === undefined) {
+                OpenIdRecipe.instance = new OpenIdRecipe(OpenIdRecipe.RECIPE_ID, appInfo, isInServerlessEnv, config);
+                return OpenIdRecipe.instance;
+            } else {
+                throw new Error("OpenId recipe has already been initialised. Please check your code for bugs.");
+            }
+        };
+    }
+
+    static reset() {
+        if (process.env.TEST_MODE !== "testing") {
+            throw new Error("calling testing function in non testing env");
+        }
+        OpenIdRecipe.instance = undefined;
+    }
+
+    getAPIsHandled = (): APIHandled[] => {
+        return [
+            {
+                method: "get",
+                pathWithoutApiBasePath: new NormalisedURLPath(GET_DISCOVERY_CONFIG_URL),
+                id: GET_DISCOVERY_CONFIG_URL,
+                disabled: this.apiImpl.getOpenIdDiscoveryConfigurationGET === undefined,
+            },
+            ...this.jwtRecipe.getAPIsHandled(),
+        ];
+    };
+    handleAPIRequest = async (
+        id: string,
+        req: BaseRequest,
+        response: BaseResponse,
+        path: normalisedURLPath,
+        method: HTTPMethod
+    ): Promise<boolean> => {
+        let apiOptions: APIOptions = {
+            recipeImplementation: this.recipeImplementation,
+            config: this.config,
+            recipeId: this.getRecipeId(),
+            req,
+            res: response,
+        };
+
+        if (id === GET_DISCOVERY_CONFIG_URL) {
+            return await getOpenIdDiscoveryConfiguration(this.apiImpl, apiOptions);
+        } else {
+            return this.jwtRecipe.handleAPIRequest(id, req, response, path, method);
+        }
+    };
+    handleError = async (error: STError, request: BaseRequest, response: BaseResponse): Promise<void> => {
+        if (error.fromRecipe === OpenIdRecipe.RECIPE_ID) {
+            throw error;
+        } else {
+            return await this.jwtRecipe.handleError(error, request, response);
+        }
+    };
+    getAllCORSHeaders = (): string[] => {
+        return [];
```

Comment:
Needs to do call jwtRecipe's `getAllCORSHeaders` too

Reply:
Done